ragtime.jdbc/src/ragtime/jdbc.clj

```diff
@@ -17,10 +17,18 @@
                         [[:id "varchar(255)"]
                          [:created_at "varchar(32)"]]))
 
-(defn- ensure-migrations-table-exists [db-spec migrations-table]
-  (try
-    (sql/execute! db-spec [(migrations-table-ddl migrations-table)])
-    (catch SQLException _)))
+(defn- table-exists?
+  [connection table-name]
+  (when connection
+    (let [results (.getTables (.getMetaData connection) nil nil table-name nil)
+          n (.next results)]
+      (and n (= (.getString results 3) table-name)))))
+
+(defn- ensure-migrations-table-exists
+  [{:keys [connection] :as db-spec} migrations-table]
+  ;; basically everything should be upper-cased except postgres
+  (when-not (table-exists? connection (.toUpperCase migrations-table))
```

Comment:
Is it expensive? How many tables does an average project have, and how long does it take to iterate through them? We should be careful about making decisions on performance without data to back them up.

So instead, let's go for the most straightforward solution. If it turns out this impacts the runtime of the migrations significantly in real world projects, then we can easily add a cache or change the design in some way.